aiida_common_workflows/cli/launch.py

```diff
@@ -78,3 +79,131 @@ def cmd_relax(plugin, structure, protocol, relaxation_type, threshold_forces, th
+@options.PROTOCOL(type=click.Choice(['fast', 'moderate', 'precise']), default='fast')
+@options.RELAXATION_TYPE()
+@options.THRESHOLD_FORCES()
+@options.THRESHOLD_STRESS()
+@options.DAEMON()
+@click.option('--show-engines', is_flag=True, help='Show information on the required calculation engines.')
+def cmd_eos(plugin, structure, protocol, relaxation_type, threshold_forces, threshold_stress, daemon, show_engines):
+    """Relax a crystal structure using the common relax workflow for one of the existing plugin implementations.
+
+    The command will automatically try to find and load the codes that are required by the plugin workflow. If no code
+    is installed for at least one of the calculation engines, the command will fail. Use the `--show-engine` flag to
+    display the required calculation engines for the selected plugin workflow.
+    """
+    # pylint: disable=too-many-locals
+    from aiida.orm import QueryBuilder, Code
+    from aiida_common_workflows.plugins import get_entry_point_name_from_class
+    from aiida_common_workflows.workflows.eos import EquationOfStateWorkChain
+
+    process_class = load_workflow_entry_point('relax', plugin)
+    generator = process_class.get_inputs_generator()
+
+    if not generator.is_valid_protocol(protocol):
+        protocols = generator.get_protocol_names()
+        process_class_name = process_class.__name__
+        message = f'`{protocol}` is not implemented by `{process_class_name}` workflow: choose one of {protocols}'
+        raise click.BadParameter(message, param_hint='protocol')
+
+    if show_engines:
+        for engine in generator.get_calc_types():
+            schema = generator.get_calc_type_schema(engine)
+            click.secho(engine, fg='red', bold=True)
+            click.echo('Required code plugin: {}'.format(schema['code_plugin']))
+            click.echo('Engine description:   {}'.format(schema['description']))
+
+        return
+
+    engines = {}
+
+    for engine in generator.get_calc_types():
+        schema = generator.get_calc_type_schema(engine)
+        engines[engine] = {
+            'options': {
+                'resources': {
+                    'num_machines': 1
+                },
+                'max_wallclock_seconds': 86400,
+            }
+        }
+        code_plugin = schema['code_plugin']
+        builder = QueryBuilder().append(Code, filters={'attributes.input_plugin': code_plugin})
+
+        code = builder.first()
+
+        if code is None:
+            raise click.UsageError(f'could not find a configured code for the plugin `{code_plugin}`.')
+
+        engines[engine]['code'] = code[0]
+
+    builder = generator.get_builder(structure, engines, protocol, relaxation_type, threshold_forces, threshold_stress)
+    inputs = {
+        'structure': structure,
+        'sub_process_class': get_entry_point_name_from_class(process_class).name,
+        'sub_process': {
+            **builder._inputs(prune=True)  # pylint: disable=protected-access
```

Comment:
Maybe it makes sense to make available to the user as well a method that returns the inputs in a dictionary.
I'm saying to create a method in the `RelaxInputsGenerator` called for instance `get_inputs_dict` that returns `builder._inputs(prune=True)`. It is useful in my opinion. Sometimes one prefers to have the dictionary of inputs instead of the builder.

Reply:
Good point, I will add that. This way they also don't need to remember the `._inputs(prune=True)` idiom.

Reply:
Actually, coming back to this, there is a reason we made it a method starting with an underscore, because otherwise it would be polluting the namespace. Currently, all public attributes of a builder reflect an input port. If we add `get_input_dict` people can confuse it, during tab-completion, for an input port.

Reply:
I meant to add a method `get_input_dict` to the `generator` not to the `builder`! I would probably just add to `RelaxInputsGenerator`:

     def get_input_dict(same signature of the get_builder):
           builder = self.get_builder(same signature of the get_builder)
           return builder._inputs(prune=True)